versions/protocol-2-x.md

```diff
@@ -1465,6 +1466,123 @@ _Response_
 * result: `SymbolInformation[]` as defined above.
 * error: code and message set in case an exception happens during the workspace symbol request.
 
+#### <a name="workspace_documentation"></a>Workspace Documentation Request
+
+The workspace documentation request is sent from the client to the server to list project-wide documentation.
+
+_Request_
+* method: 'workspace/documentation'
+* params: `WorkspaceDocumentationParams` defined as follows:
+```typescript
+/**
+ * The parameters of a Workspace Documentation Request.
+ */
+interface WorkspaceDocumentationParams {
+	/**
+	 * An optional non-empty query string, to be matched against the `name` of
+	 * `Documentation` in the workspace. For example, a user may query `HTTP Request`
+	 * and the server may respond with the `Documentation` named `Request`
+	 * which is itself a child of `Documentation` named `HTTP`.
+	 */
+	query?: string;
+
+	/**
+	 * An optional non-empty query string for the `Documentation` with the specified ID.
+	 */
+	queryID?: number | string;
+}
+```
+
+_Response_
+* result: `Documentation?` defined as follows:
+```typescript
+/**
+ * Represents documentation for a programming construct (variable, function,
+ * etc.) or group of programming constructs in a workspace (library, package,
+ * crate, module, etc.)
+ *
+ * The language server defines the exact structure of the documentation in a
+ * way that makes sense for the specific language and concepts being described.
+ */
+interface Documentation {
+	/**
+	 * A unique identifier for this `Documentation`. It exists to allow one
+	 * piece of `Documentation` to link to another piece of `Documentation` via
+	 * `InteractiveString`.
+	 */
+	id: number | string;

```

Comment:
Good question. My thinking was the `label` field would be used for URLs, but I recognize that might not be unique and so not suitable.


I'll see about changing this to a string, and adding a note that clients may use it in URLs and so it may be nice to make it human-readable.